build.sbt

```diff
@@ -864,6 +881,25 @@ lazy val testsCommonSettings = Def.settings(
   )
 )
 
+lazy val javaVersionBasedTestSources = Def.settings(
+  Test / unmanagedSourceDirectories ++= {
+    // unit-tests/native/src/test
+    val testDir = (Test / baseDirectory).value
```

Comment:
It is not correct, baseDirectory would point to `unit-tests/native` or `unit-tests/jvm`

Reply:
Sorry, left the comment from before.